### PR TITLE
Make back button behaviour more consistant

### LINKED ISSUE #1750: Back button disappears after navigating back to search results
### What Happened?

It looks like the logic for hiding the back button is bad, as reported downstream: https://github.com/pop-os/shop/issues/300

### Steps to Reproduce

1. Search for something in AppCenter
2. Click a result
3. Go back to the results
4. Notice the back button is gone

### Expected Behavior

The back button should always be visible on the search results view

### OS Version

6.x (Odin)

### Software Version

Latest release (I have run all updates)

### Log Output

_No response_

### Hardware Info

_No response_

### PR DESCRIPTION
**Pull request by 13r0ck**
Fixes #1750 and #1751

### CODE REVIEW

**Review by cassidyjames — CHANGES_REQUESTED** [review]
@13r0ck thanks for submitting this! However, I'm still seeing some weird behavior here. When navigating from Home → Category (e.g. Accessories) → Search, the back button here still says "Home," and takes you home. However, I'd expect it to back out of the search and bring you back to the category.

If that's a much more invasive change I'd consider merging this as-is to fix the listed issues, but it still feels off, so I wonder if we can take the time to make it right.

**Comment by 13r0ck** [comment]
Ya it will take me a bit to get around to that, but I will make it happen. :smile:

**Comment by 13r0ck** [comment]
Nevermind I fixed it today

File: src/MainWindow.vala [hunk 1]
```diff
@@ -104,6 +104,7 @@ public class AppCenter.MainWindow : Hdy.ApplicationWindow {
         search_view.package_selected.connect (package_selected);
         search_view.subview_entered.connect (view_opened);
         search_view.home_return_clicked.connect (show_homepage);
+        search_view.category_return_clicked.connect(show_category);
```
Comment (cassidyjames) [inline src/MainWindow.vala:107]: Lint: space before paren

```suggestion
        search_view.category_return_clicked.connect (show_category);
```

File: src/Views/SearchView.vala [hunk 2]
```diff
@@ -51,17 +52,15 @@ public class AppCenter.Views.SearchView : AbstractView {
                 viewing_package = false;
 
                 if (current_category != null) {
-                    subview_entered (_("Search Apps"), true, current_category.name);
+                    subview_entered (_(current_category.name), true, current_category.name);
```
Comment (cassidyjames) [inline src/Views/SearchView.vala:55]: If I'm reading this correctly, this should not be marked as translatable 

```suggestion
                    subview_entered (current_category.name, true, current_category.name);
```

File: src/Views/SearchView.vala [hunk 3]
```diff
@@ -84,7 +83,7 @@ public class AppCenter.Views.SearchView : AbstractView {
         }
 
         if (current_category != null) {
-            subview_entered (_("Search Apps"), true, current_category.name);
+            subview_entered (_(current_category.name), true, current_category.name);
```
Comment (cassidyjames) [inline src/Views/SearchView.vala:86]: Same as above

```suggestion
            subview_entered (current_category.name, true, current_category.name);
```

**Review by cassidyjames — CHANGES_REQUESTED** [review]
Thanks again for this! I've taken an initial look and noticed some small fixes. I'll give it some testing but I think I'd like a more thorough review from @elementary/desktop-developers.

**Comment by cassidyjames** [comment]
I can confirm that this seems to fix the back button labeling and behavior, though!

**Comment by 13r0ck** [comment]
Ah I didn't know that `_()` meant the string was translatable. Thank you for the review and teaching me!

**Review by zeebok — COMMENTED** [review]
I think the code here looks good. I do wonder why some kind of stack isn't used to track previously visited views but that is something to explore another time. @cassidyjames if your tests are showing green I think this is good to go!

**Review by cassidyjames — APPROVED** [review]

**Pull request merged**